test/unit/specs/utils/getBrowser.spec.js

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import getBrowser, {
+  EDGE,
+  CHROME,
+  FIREFOX,
+  IE,
+  SAFARI,
+  UNKNOWN
+} from "../../../../src/utils/getBrowser";
+
+const userAgentsByBrowser = {
+  [EDGE]: [
+    "Mozilla/5.0 (Windows NT 10.0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/42.0.2311.135 Safari/537.36 Edge/12.10136",
+    "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/64.0.3282.140 Safari/537.36 Edge/17.17134"
+  ],
+  // Chrome on iOS will not be detected as Chrome. That is because Chrome on
+  // iOS is just Safari wrapped in Chrome.
+  [CHROME]: [
+    "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/78.0.3904.87 Safari/537.36",
+    "Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/66.0.3359.117 Safari/537.36"
+  ],
+  [FIREFOX]: [
+    "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.14; rv:69.0) Gecko/20100101 Firefox/69.0",
+    "Mozilla/5.0 (Windows NT 6.1; WOW64; rv:54.0) Gecko/20100101 Firefox/54.0"
+  ],
+  [IE]: [
+    "Mozilla/5.0 (Windows NT 10.0; WOW64; Trident/7.0; .NET4.0C; .NET4.0E; Tablet PC 2.0; rv:11.0) like Gecko",
+    "Mozilla/5.0 (Windows NT 6.3; Trident/7.0; rv:11.0) like Gecko"
+  ],
+  [SAFARI]: [
+    "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/12.0 Safari/605.1.15",
+    "Mozilla/5.0 (iPhone; CPU iPhone OS 12_2 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/13 Mobile/15E148 Safari/604.1"
+  ],
+  // Note that browsers like Opera will usually be reported as something like
+  // Chrome instead of Unknown because they contain other browser names in their
+  // user agent strings. For example, this one from Opera:
+  // Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/78.0.3904.97 Safari/537.36 OPR/64.0.3417.92
```

Comment:
This note is important to understand. Due to how complicated user agent strings are now with practically every browser trying to mimic other browsers, some browsers that we aren't specifically trying to detect will end up being detected as one of our supported browsers. Unless we add regexes for more browsers, I think this is about the best we can do.

Reply:
Thank you @Aaronius. We can bring this up with Justin again and see if he'd like to be more granular.

Reply:
What do you think about using a library for this?

Reply:
The libraries I looked at all try to do a lot more than what we need and would end up bloating the library. If there's one you suggest though, I think we should consider it.

Reply:
I do think some of the libraries have much more specific regexes though that could keep browsers like Opera in the "unknown" bucket. I'm going to try out some of them.

Reply:
I was looking at detect-browser from a quick google search. It would add some bloat to alloy, but having to keep this up to date on our own is a big investment.

Reply:
I've been looking at detect-browser as well. Some of its regular expressions will keep browsers like Opera (but not Brave) in the Unknown bucket rather than the Chrome bucket, which is nice. I'm going to copy some of its regular expressions (I'll open a new PR for that work), but I don't feel like using the library directly is worth the cost.